TAO/tao/Valuetype/ValueBase.h

```diff
@@ -319,37 +319,27 @@ namespace CORBA
    *
    * Default mix-in for reference count of a valuetype.
    */
-  class TAO_Valuetype_Export DefaultValueRefCountBase
-    : public virtual ValueBase
+  class TAO_Valuetype_Export DefaultValueRefCountBase : public virtual ValueBase
   {
   public:
-    virtual void _add_ref ();
-    virtual void _remove_ref ();
-    virtual CORBA::ULong _refcount_value ();
-
-    /// The _tao variants are inline for fast access from T_var
-    /// (if valuetype T is compiled with optimization for that.) %! (todo)
-    void _tao_add_ref ();
-    void _tao_remove_ref ();
-    CORBA::ULong _tao_refcount_value () const;
+    void _add_ref () override;
+    void _remove_ref () override;
+    CORBA::ULong _refcount_value () override;
 
   protected:
-    DefaultValueRefCountBase ();
+    DefaultValueRefCountBase () = default;
     DefaultValueRefCountBase (const DefaultValueRefCountBase &) = delete;
-    virtual ~DefaultValueRefCountBase ();
+    DefaultValueRefCountBase (DefaultValueRefCountBase &&) = delete;
+    ~DefaultValueRefCountBase () override = default;
 
   private:
     DefaultValueRefCountBase &operator= (const DefaultValueRefCountBase &) = delete;
+    DefaultValueRefCountBase &operator= (DefaultValueRefCountBase &&) = delete;
```

Comment:
The deleted copy/move constructors are protected while the deleted copy/move assignment operators are private.  It would be good to be consistent -- I suppose private is preferred?

Reply:
Moved to private